Conceivable sizes

dimwit105 opened the pull request:
Closes DFHack/DFHack#4506

Adjusts the size line to compare units to something more reasonable than cubic centimeters. Will compare small things to cats, most animals and humanoids to dwarves, and particularly large animals to elephants. 

Also adds a line to show that particular unit's size modifiers.

myk002 commented on gui/unit-info-viewer.lua:
please add a line about this improvement in `changelog.txt`

myk002 commented:
pre-commit.ci autofix

myk002 commented on gui/unit-info-viewer.lua:
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
+end
+
+local function get_size_compared_to_median(unit)
+    local size_normal = unit.body.size_info.size_base
```
unused variable?

myk002 commented on gui/unit-info-viewer.lua:
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
 end
 
+local function get_conceivable_comparison(unit, comparison_unit)
+    --the objective here is to get a (resaonably) small number to help concieve how large a thing is. "83 dwarves" doesn't really help convey the size of an elephant much better than 5m cc, so at certain breakpoints we will use different animals
```
```suggestion
    -- the objective here is to get a (reasonably) small number to help conceive
    -- how large a thing is. "83 dwarves" doesn't really help convey the size of an
    -- elephant much better than 5m cc, so at certain breakpoints we will use
    -- different animals
```

myk002 commented on gui/unit-info-viewer.lua:
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
     return unit.body.size_info.size_cur * 10
```
is this function now unused? if so, you can remove it

myk002 commented on gui/unit-info-viewer.lua:
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
+    local size = unit.body.size_info.size_cur
+    local base_comparison_size = 6000
+    if size > base_comparison_size*20 then
+        return string.format('%.3f average elephants', size/500000)
+    elseif size > base_comparison_size*0.25 then
+        return string.format('%.3f average dwarves', size/6000)
+    else
+        return string.format('%.3f average cats', size/500)
```
I'm not sure we even need this much accuracy. How about `%.1f`?

myk002 commented on gui/unit-info-viewer.lua:
```diff
@@ -451,6 +481,7 @@ function UnitInfo:refresh(unit, width)
     add_chunk(chunks, get_ghostly_chunk(unit), width)
     add_chunk(chunks, get_dead_chunk(unit), width)
     add_chunk(chunks, get_body_chunk(unit), width)
+    add_chunk(chunks, get_average_size(unit), width)
```
maybe avoid adding this chunk if the race is dwarf, elephant, or cat. We already know the unit is average/small/large for its race; we don't need to know that the elephant weighs about as much as 1.0 elephants.

dimwit105 commented:
I've removed the extra function after checking it wasnt used anywhere else, and reformatted that long comment so it isnt a huge line anymore. 

I've adjusted the way weights are displayed, so elephants/cats arent reported as weighing 1 elephant or 1 cat, but instead display weight in dwarves. Dwarves themselves no longer give a precise weight, just the relative to average dwarfiness.

myk002 commented:
pre-commit.ci autofix

myk002 reviewed (commented):
Looks great! Could you add a line to the "Misc Improvements" section for the "Future" version in `changelog.txt`?

myk002 commented:
Also, please feel free to add yourself to https://github.com/DFHack/dfhack/blob/develop/docs/about/Authors.rst (in a separate PR)

dimwit105 commented:
The changelog has been added. I have managed to bumble through my first GitHub experience, and my journey to GitHub Wizard has begun
